fix: correct month names localization in Datepicker

LeonidShv opened the pull request:
## Description
I've fixed a part of the published issue: ["Data picker behaviour with non-English"](https://github.com/infermedica/component-library/issues/526). 

When you change the language, the months in the date picker didn’t update and showed incorrect translations.

Commits:
[fix: correct month names localization in Datepicker and replace monthNames ref with computed](https://github.com/infermedica/component-library/pull/547/commits/f8b7d24b2111d42a712909bac548180b7545e58c)
→ I've made monthNames a computed property that updates automatically when props.lang changes

The issue with the capitalization of the first letter in month names is not fixed in this PR.
I plan to address it in a separate fix after this PR is successfully merged.

## Related Issue
["Issue 526: Data picker behaviour with non-English"](https://github.com/infermedica/component-library/issues/526). 

steps to reproduce:
Go to https://component.infermedica.com/?path=/story/organisms-datepicker--full-configuration
Click on props.lang
Select any language other than en or en-us
See that nothing changes in Month tab

Closes #

## Motivation and Context
["Issue 526: Data picker behaviour with non-English"](https://github.com/infermedica/component-library/issues/526). 
When changing the language, the months in the date picker didn’t update and showed incorrect translations.


## How Has This Been Tested?
You can test it by using the DatePicker component and changing the lang prop.
Example:

```
<UiDatepicker :lang="selectedLang" />
```

## Screenshots (if appropriate):

https://github.com/user-attachments/assets/66127182-a234-4dcb-bf8b-f437d3dfdd09


## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

LeonidShv committed:
fix: correct month names localization in Datepicker and replace monthNames ref with computed

AlanLes commented:
Hello @LeonidShv! 🙇 
Both your code and the solution look good, and I'll approve that, for sure 😄 

However, I see an inconsistency in the main **Description**. 

You've written that:

> fix: correct month names localization in Datepicker
> → Added a watcher for the lang prop to update month names when the language changes.

I don't see any new watcher added. Is that up to date?

LeonidShv commented:
> fix: correct month names localization in Datepicker
> → Added a watcher for the lang prop to update month names when the language changes.

Hi @AlanLes ,
Thanks for the quick reply! 

Maybe I didn’t explain it very well.

First, I added the watcher in the first commit. Then, in the second commit, I removed this watcher as part of a small refactoring.
If needed, we can squash these commits during the PR merge. Also, if you prefer, I can squash them myself in this PR.

I’d be happy if you could share some links to your internal resources — maybe a task board, code style guide, or commit naming convention.

AlanLes commented:
@LeonidShv, yes, please squash it, and let's make this PR full-blown 😄 👍 

Regarding our internal resources, I'm afraid all our resources are, as you wrote, _internal_. Most of the docs are on our Confluence, which is restricted to Infermedica's employees. However, I believe [the docs](https://component.infermedica.com/) should be sufficient for you. The most crucial information is there. 🙂 

Thanks for your contribution! 🙇

LeonidShv commented:
@AlanLes 
I've squashed 2 commits. I hope everything is okay, and I'll wait for the approval. 
I'm always open to questions or ideas.

Regarding to the documents and code style, I will try to compare my PRs with existed closed on.

Thanks for your time : )

AlanLes commented:
@LeonidShv, I still don't see any watcher added here. 

<img width="1405" alt="image" src="https://github.com/user-attachments/assets/295afb9f-e3cc-4368-9ade-2e31de62f43a" />

LeonidShv commented:
@AlanLes 
Thank you for the updates. 
That's my mistake — I forgot to update the PR description after squashing the commits. I've updated the PR description accordingly. 

_After squashing the commits, the watcher is no longer present_

AlanLes reviewed (approved):
Awesome, the code looks good 😄 👍
@DanielHajduk98 @sonya0504 @pspaczek @mateuszkulpa, would you guys like to see and approve, or add any comments to it?

LeonidShv commented:
@AlanLes Thank you for the feedback! I hope my next PR will be clearer : )
I don't see the merge button. Could you please help merge this PR? Probably, I don't have permissions for that as an external collaborator

AlanLes commented:
@LeonidShv sure, I've approved it. Let's wait for another approval from someone from the team. @sonya0504 @DanielHajduk98 @kubajmarek @pspaczek @mateuszkulpa @jbucior? :)

kubajmarek commented:
I don't think my or Przemek's approval would be useful here, as it doesn't have much power after org removal. 😉
